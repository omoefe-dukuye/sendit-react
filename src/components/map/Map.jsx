import React, { Component } from 'react';
import { Map, Marker, Polyline, GoogleApiWrapper } from 'google-maps-react';
import PropTypes from 'prop-types';

const mapStyles = {
  width: '100%',
  height: '100%'
};

export class MapContainer extends Component {
  render() {
    const { initialCenter, destination, pickupLocation, google } = this.props;
    return (
      <Map
        zoom={14}
        style={mapStyles}
        initialCenter={initialCenter}
        google={google}
      >
        <Marker
          animation= {google.maps.Animation.BOUNCE}
          name={'Current Location'}
          position={initialCenter} />
        <Marker
          name={'Pickup Location'}
          position={pickupLocation} />
        <Marker />
        <Marker
          name={'Destination'}
          position={destination} />
        <Marker />
        <Polyline
          path={[pickupLocation, initialCenter]}
          strokeColor='hsl(120, 50%, 75%)'
          strokeOpacity={0.8}
          strokeWeight={2} />
        <Polyline
          path={[initialCenter, destination]}
          strokeColor='hsl(0, 50%, 60%)'
          strokeOpacity={0.8}
          strokeWeight={2} />
      </Map>
    );
  }
}

MapContainer.propTypes = {
  google: PropTypes.object,
  initialCenter: PropTypes.object,
  pickupLocation: PropTypes.object,
  destination: PropTypes.object,
}

export default GoogleApiWrapper(({ initialCenter }) => ({
  apiKey: process.env.GOOGLE_API_KEY,
  initialCenter
}))(MapContainer);
