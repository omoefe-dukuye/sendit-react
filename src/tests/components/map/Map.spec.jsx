import React from 'react';
import { shallow } from 'enzyme';
import Map from '../../../components/map';

test('Should render static component', () => {
  const wrapper = shallow(<Map />);
  expect(wrapper).toMatchSnapshot();
});
