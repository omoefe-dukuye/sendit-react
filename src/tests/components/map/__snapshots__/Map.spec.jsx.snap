// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `<DefaultLoadingContainer />`;
